core/services/offchainreporting/config_overrider.go

```diff
@@ -0,0 +1,150 @@
+	chDone    chan struct{}
+
+	mu sync.Mutex
+}
+
+// InitialHibernationStatus - hibernation state set until the first successful update from the chain
+const InitialHibernationStatus = false
+
+func NewConfigOverriderImpl(
+	logger *logger.Logger,
+	contractAddress ethkey.EIP55Address,
+	flags *ContractFlags,
+	pollInterval time.Duration,
+) (*ConfigOverriderImpl, error) {
+
+	if !flags.ContractExists() {
+		return nil, errors.Errorf("OCRConfigOverrider: Flags contract instance is missing, the contract does not exist: %s. "+
+			"Please create the contract or remove the FLAGS_CONTRACT_ADDRESS configuration variable", contractAddress.Address())
+	}
+
+	addressBig := contractAddress.Big()
+	addressInt64 := addressBig.Mod(addressBig, big.NewInt(3600)).Int64()
+	deltaC := 23*time.Hour + time.Duration(addressInt64)*time.Second
+
+	ctx, cancel := context.WithCancel(context.Background())
+	co := ConfigOverriderImpl{
+		utils.StartStopOnce{},
+		logger,
+		flags,
+		contractAddress,
+		pollInterval,
+		time.Now(),
+		InitialHibernationStatus,
+		deltaC,
+		ctx,
+		cancel,
+		make(chan struct{}),
+		sync.Mutex{},
+	}
+
+	return &co, nil
+}
+
+func (c *ConfigOverriderImpl) Start() error {
+	return c.StartOnce("OCRConfigOverrider", func() (err error) {
+		if err := c.updateFlagsStatus(); err != nil {
+			c.logger.Errorw("OCRConfigOverrider: Error updating hibernation status at OCR job start. Will default to not hibernating, until next successful update.", "err", err)
+		}
+
+		go c.eventLoop()
+		return nil
+	})
+}
+
+func (c *ConfigOverriderImpl) Close() error {
+	return c.StopOnce("OCRContractTracker", func() error {
+		c.ctxCancel()
+		<-c.chDone
+		return nil
+	})
+}
+
+func (c *ConfigOverriderImpl) eventLoop() {
+	defer close(c.chDone)
+	ticker := time.NewTicker(utils.WithJitter(c.flagsUpdateInterval))
+	defer ticker.Stop()
+	for {
+		select {
+		case <-c.ctx.Done():
+			return
+		case <-ticker.C:
+			if err := c.updateFlagsStatus(); err != nil {
+				c.logger.Errorw("OCRConfigOverrider: Error updating hibernation status", "err", err)
+			}
+		}
+	}
+}
+
+func (c *ConfigOverriderImpl) updateFlagsStatus() error {
+	isFlagLowered, err := c.flags.IsLowered(c.contractAddress.Address())
+	if err != nil {
+		return err
+	}
+	shouldHibernate := !isFlagLowered
+
+	c.mu.Lock()
+	defer c.mu.Unlock()
+
+	if !c.isHibernating && shouldHibernate {
+		c.logger.Infow("OCRConfigOverrider: Setting hibernation state to 'hibernating'", "elapsedSinceLastChange", time.Since(c.lastStateChangeTimestamp))
+		c.lastStateChangeTimestamp = time.Now()
+	}
+
+	if c.isHibernating && !shouldHibernate {
+		c.logger.Infow("OCRConfigOverrider: Setting hibernation state to 'not hibernating'", "elapsedSinceLastChange", time.Since(c.lastStateChangeTimestamp))
+		c.lastStateChangeTimestamp = time.Now()
+	}
+
+	c.isHibernating = shouldHibernate
+	return nil
+}
+
+func (c *ConfigOverriderImpl) ConfigOverride() *ocrtypes.ConfigOverride {
+	if c == nil {
+		return nil
+	}
```

Comment:
I can't see yet how this was happening without this check for nil, but TestIntegration_OCR was failing with:
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x51081ae]

goroutine 10326 [running]:
github.com/smartcontractkit/chainlink/core/services/offchainreporting.(*ConfigOverriderImpl).ConfigOverride(0x0, 0x0)
        /Users/piotr/go/src/chainlink/core/services/offchainreporting/config_overrider.go:135 +0xae
github.com/smartcontractkit/libocr/offchainreporting/internal/managed.ConfigOverriderWrapper.ConfigOverride(...)
        /Users/piotr/go/pkg/mod/github.com/smartcontractkit/libocr@v0.0.0-20210820145132-3c090fd196cc/offchainreporting/internal/managed/config_overrider.go:16
github.com/smartcontractkit/libocr/offchainreporting/internal/protocol.(*reportGenerationState).shouldReport(0xc001cdb440, 0xc00400ff80, 0x3, 0x3, 0x3)
```
where the wrapper on libocr side should prevent against it..